File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    const visited = Array(n).fill(false);
    let cnt = 0;
    
    const dfs = i => {
        visited[i] = true;
        for (let j = 0; j < n; j++) {
            if (visited[j] || !computers[i][j]) continue;
            dfs(j);
        }
    }
    
    for (let i = 0; i < n; i++) {
        if (visited[i]) continue;
        dfs(i);
        cnt++;
    }
    
    return cnt;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    net = 0
    visited = [False] * n

    def bfs(s):
        q = deque([s])
        visited[s] = True
        
        while q:
            cur = q.popleft()

            for i, connect in enumerate(computers[cur]):
                if visited[i] or not connect: continue
                
                visited[i] = True
                q.append(i)
    
    for i in range(n):
        if visited[i]: continue
        
        bfs(i)
        net += 1
    
    return net